pkg/i2gw/providers/ingressnginx/server_alias.go

```diff
@@ -0,0 +1,164 @@
+			for _, alias := range aliases {
+				// Check if alias already exists for this gateway
+				exists := slices.Contains(existingAliases, alias)
+				if !exists {
+					serverAliasesByGateway[gatewayKey] = append(serverAliasesByGateway[gatewayKey], alias)
+				}
+			}
+
+			notify(notifications.InfoNotification,
+				fmt.Sprintf("parsed server-alias annotation with %d aliases: %s", len(aliases), strings.Join(aliases, ", ")), &ingress)
+		}
+	}
+
+	// Process each gateway that has server aliases
+	for gatewayKey, aliases := range serverAliasesByGateway {
+		gatewayContext, exists := ir.Gateways[gatewayKey]
+		if !exists {
+			continue // Skip if the gateway doesn't exist in the IR
+		}
+
+		// Find an existing listener to base the new ones on (use the first HTTP/HTTPS listener)
+		var baseListener *gatewayv1.Listener
+		for i, listener := range gatewayContext.Spec.Listeners {
+			if listener.Protocol == gatewayv1.HTTPProtocolType || listener.Protocol == gatewayv1.HTTPSProtocolType {
+				baseListener = &gatewayContext.Spec.Listeners[i]
+				break
+			}
+		}
+
+		if baseListener != nil {
+			// Create additional listeners for each server alias
+			for _, alias := range aliases {
+				hostname := gatewayv1.Hostname(alias)
+
+				// Create a new listener based on the existing one but with the alias hostname
+				aliasListener := gatewayv1.Listener{
+					Name:     gatewayv1.SectionName(strings.ReplaceAll(alias, ".", "-")),
+					Hostname: &hostname,
+					Protocol: baseListener.Protocol,
+					Port:     baseListener.Port,
+					TLS:      baseListener.TLS,
+				}
+
+				// Add the new listener to the gateway
+				gatewayContext.Spec.Listeners = append(gatewayContext.Spec.Listeners, aliasListener)
+
+				notify(notifications.InfoNotification,
+					fmt.Sprintf("added server alias listener for hostname: %s", alias),
+					&gatewayContext.Gateway)
+			}
+
+			// Update the gateway context in the IR
+			ir.Gateways[gatewayKey] = gatewayContext
+		}
+
+		// Also update HTTPRoutes to include the server aliases as additional hostnames
+		for routeKey, routeContext := range ir.HTTPRoutes {
+			if routeKey.Namespace == gatewayKey.Namespace {
+				// Add server aliases as additional hostnames to the HTTPRoute
+				for _, alias := range aliases {
+					hostname := gatewayv1.Hostname(alias)
+					// Check if this hostname already exists
+					exists := slices.Contains(routeContext.Spec.Hostnames, hostname)
+
+					if !exists {
+						routeContext.Spec.Hostnames = append(routeContext.Spec.Hostnames, hostname)
+						notify(notifications.InfoNotification,
+							fmt.Sprintf("added server alias hostname to HTTPRoute: %s", alias),
+							&routeContext.HTTPRoute)
+
+						// Update the route context in the IR
+						ir.HTTPRoutes[routeKey] = routeContext
+					}
+				}
+			}
+		}
+	}
+
+	return errs
+}
+
+// getIngressClass returns the ingress class for the ingress
+func getIngressClass(ingress networkingv1.Ingress) string {
```

Comment:
You can probably use https://github.com/kubernetes-sigs/ingress2gateway/blob/312bd894ba8553c566eecf2fecbbb34451ae6b29/pkg/i2gw/providers/common/utils.go#L31 instead of duplicating this function